### PR TITLE
Enable the running of performance tests: SDK:release/2.0.0

### PR DESCRIPTION
**Pull request by ghost**
Enable the running of performance tests: 
- commandLine: "build.cmd -perf"
- optional properties: 'PerfIterations' 'PerfOutputDirectory' 'PerfCollectionType'
- update RepoToolset version to a version that contains the performance targets
- first draft 'perf.groovy'
- rename: "Microsoft.NET.Perf.Tests" to "Microsoft.NET.PerformanceTests"

@DrewScoggins

### CODE REVIEW

**Comment by johnbeisner** [comment]
@dsplaisted

File: perf.groovy [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+def static getBuildJobName(def configuration, def os) {
+    return configuration.toLowerCase() + '_' + os.toLowerCase()
+}
+
+// Setup perflab tests runs
+[true, false].each { isPR ->
+    ['Windows_NT'].each { os ->
+        ['x64', 'x86'].each { arch ->
+            def architecture = arch
+            def jobName = "perf_perflab_${os}_${arch}"
```
Comment (DrewScoggins) [inline perf.groovy]: Probably want to update the name of the job to better reflect the SDK. Something like SDK_Perf_${os}_${arch}

**Comment by johnbeisner** [comment]
@DrewScoggins
@tannergooding 
Can I get a code review on this please.

File: perf.groovy [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+                    batchFile("\"${perfWorkingDirectory}\\nuget.exe\" install Microsoft.BenchView.JSONFormat -Source http://benchviewtestfeed.azurewebsites.net/nuget -OutputDirectory \"${perfWorkingDirectory}\" -Prerelease -ExcludeVersion")
+                    // Do this here to remove the origin but at the front of the branch name as this is a problem for BenchView
+                    //    we have to do it all as one statement because cmd is called each time and we lose the set environment variable
+                    batchFile("if \"%GIT_BRANCH:~0,7%\" == \"origin/\" (set \"GIT_BRANCH_WITHOUT_ORIGIN=%GIT_BRANCH:origin/=%\") else (set \"GIT_BRANCH_WITHOUT_ORIGIN=%GIT_BRANCH%\")\n" +
+                    "set \"BENCHVIEWNAME=${benchViewName}\"\n" +
+                    "set \"BENCHVIEWNAME=%BENCHVIEWNAME:\"=\"\"%\"\n" +
+                    "py \"${perfWorkingDirectory}\\Microsoft.BenchView.JSONFormat\\tools\\submission-metadata.py\" --name \"%BENCHVIEWNAME%\" --user-email \"dotnet-bot@microsoft.com\" --output \"${perfWorkingDirectory}\"\n" +
```
Comment (DrewScoggins) [inline perf.groovy]: I am fairly certain that when you use the output argument you need to also specify the file name, not just the out directory.

File: perf.groovy [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+                    "py \"${perfWorkingDirectory}\\Microsoft.BenchView.JSONFormat\\tools\\build.py\" git --branch %GIT_BRANCH_WITHOUT_ORIGIN% --type ${runType} --output \"${perfWorkingDirectory}\"")
+                    batchFile("py \"%WORKSPACE%\\Microsoft.BenchView.JSONFormat\\tools\\machinedata.py\" --output \"${perfWorkingDirectory}\"")
+
+			        // Build solution and run the performance tests
+				    batchFile("\"%WORKSPACE%\\build.cmd\" -perf /p:PerfIterations=10 /p:PerfOutputDirectory=\"${perfWorkingDirectory}\" /p:PerfCollectionType=stopwatch")
+
+			       //Create submission json and upload to Benchview
+				   batchFile("for /f \"tokens=*\" %%a in ('dir /b/a-d ${perfWorkingDirectory}\\*.xml') do (py \"${perfWorkingDirectory}\\Microsoft.BenchView.JSONFormat\\tools\\measurement.py xunitscenario \"${perfWorkingDirectory}\\%%a\" --better desc --output \"${perfWorkingDirectory}\")")
```
Comment (DrewScoggins) [inline perf.groovy]: Make sure that you add --append or you will overwrite the measurements

File: perf.groovy (hunk 1 above)
Comment (johnbeisner) [reply perf.groovy]: Done...

File: perf.groovy [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+                   "--machine-data \"${perfWorkingDirectory}\\machinedata.json\"" +
+                   "--metadata \"${perfWorkingDirectory}\\submission-metadata.json\"" +
+                   "--group \"SDK Perf Tests\"" +
+                   "--type \"${runType}\"" +
+                   "--config-name \"${configuration}\"" +
+                   "--config Configuration \"${configuration}\"" +
+                   "--architecture \"${arch}\"" +
+                   "--machinepool \"perfsnake\"" +
+				   "--output \"${perfWorkingDirectory}\"")
+                   batchFile("py \"${perfWorkingDirectory}\\Microsoft.BenchView.JSONFormat\\tools\\upload.py \"${perfWorkingDirectory}\submission.json\" --container stability")
```
Comment (DrewScoggins) [inline perf.groovy]: The container name should be coreclr

File: perf.groovy (hunk 3 above)
Comment (johnbeisner) [reply perf.groovy]: Done...

File: perf.groovy [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+                }
+            }
+
+            if (isPR) {
+                TriggerBuilder builder = TriggerBuilder.triggerOnPullRequest()
+                builder.setGithubContext("${os} ${arch} SDK Perf Tests")
+
+                builder.triggerOnlyOnComment()
+			    //Phrase is "test Windows_NT x64 perf"
+                builder.setCustomTriggerPhrase("(?i).*test\\W+${os}\\W+${arch}\\W+perf.*")
```
Comment (DrewScoggins) [inline perf.groovy]: This should include something about SDK in the trigger phrase

File: perf.groovy (hunk 2 above)
Comment (johnbeisner) [reply perf.groovy]: Done...

File: perf.groovy (hunk 4 above)
Comment (johnbeisner) [reply perf.groovy]: Done...

File: perf.groovy (hunk 5 above)
Comment (johnbeisner) [reply perf.groovy]: Done...

File: build/build.ps1 [hunk 6]
```diff
@@ -34,6 +35,7 @@ function Print-Usage() {
     Write-Host "  -rebuild                Rebuild solution"
     Write-Host "  -deploy                 Deploy built VSIXes"
     Write-Host "  -test                   Run all unit tests in the solution"
+	Write-Host "  -perf                   Run all performance tests in the solution"
```
Comment (dsplaisted) [inline build/build.ps1]: Use spaces instead of tabs for indentation

File: build/build.sh [hunk 7]
```diff
@@ -45,6 +46,7 @@ while [[ $# -gt 0 ]]; do
       echo "  --build                  Build solution"
       echo "  --rebuild                Rebuild solution"
       echo "  --test                   Run all unit tests in the solution"
+	  echo "  --perf                   Run all performance tests in the solution"
```
Comment (dsplaisted) [inline build/build.sh]: Use spaces instead of tabs

File: build/build.sh [hunk 8]
```diff
@@ -92,6 +94,10 @@ while [[ $# -gt 0 ]]; do
     --test)
       test=true
       shift 1
+      ;;
+	--Perf)
+      pack=true
```
Comment (dsplaisted) [inline build/build.sh]: Shouldn't this be `perf=true`?  Looks like a copy/paste error.

File: perf.groovy [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+                    //    we have to do it all as one statement because cmd is called each time and we lose the set environment variable
+                    batchFile("if \"%GIT_BRANCH:~0,7%\" == \"origin/\" (set \"GIT_BRANCH_WITHOUT_ORIGIN=%GIT_BRANCH:origin/=%\") else (set \"GIT_BRANCH_WITHOUT_ORIGIN=%GIT_BRANCH%\")\n" +
+                    "set \"BENCHVIEWNAME=${benchViewName}\"\n" +
+                    "set \"BENCHVIEWNAME=%BENCHVIEWNAME:\"=\"\"%\"\n" +
+                    "py \"${perfWorkingDirectory}\\Microsoft.BenchView.JSONFormat\\tools\\submission-metadata.py\" --name \"%BENCHVIEWNAME%\" --user-email \"dotnet-bot@microsoft.com\" --output \"${perfWorkingDirectory}\\submission-metadata.json\"\n" +
+                    "py \"${perfWorkingDirectory}\\Microsoft.BenchView.JSONFormat\\tools\\build.py\" git --branch %GIT_BRANCH_WITHOUT_ORIGIN% --type ${runType} --output \"${perfWorkingDirectory}\\build.json\"")
+                    batchFile("py \"%WORKSPACE%\\Microsoft.BenchView.JSONFormat\\tools\\machinedata.py\" --output \"${perfWorkingDirectory}\\machinedata.json\"")
+
+			        // Build solution and run the performance tests
```
Comment (dsplaisted) [inline perf.groovy]: Use spaces instead of tabs

**Review by dsplaisted — APPROVED** [review]

File: build/build.ps1 (hunk 6 above)
Comment (johnbeisner) [reply build/build.ps1]: Done...

File: build/build.sh (hunk 7 above)
Comment (johnbeisner) [reply build/build.sh]: Done...

File: build/build.sh (hunk 8 above)
Comment (johnbeisner) [reply build/build.sh]: Yup, done...

**Pull request merged**